crates/modelardb_server/src/storage/types.rs

```diff
@@ -60,63 +60,80 @@ impl MemoryPool {
         uncompressed_memory_in_bytes: usize,
         compressed_memory_in_bytes: usize,
     ) -> Self {
-        // unwrap() is safe as i64::MAX is 8192 PiB and the value is from ConfigurationManager.
         Self {
             wait_for_ingested_memory: Condvar::new(),
             remaining_ingested_memory_in_bytes: Mutex::new(
-                ingested_memory_in_bytes.try_into().unwrap(),
+                ingested_memory_in_bytes
+                    .try_into()
+                    .expect("ingested_memory_in_bytes should be less than 8192 PiB."),
             ),
             wait_for_uncompressed_memory: Condvar::new(),
             remaining_uncompressed_memory_in_bytes: Mutex::new(
-                uncompressed_memory_in_bytes.try_into().unwrap(),
+                uncompressed_memory_in_bytes
+                    .try_into()
+                    .expect("uncompressed_memory_in_bytes should be less than 8192 PiB."),
             ),
             remaining_compressed_memory_in_bytes: Mutex::new(
-                compressed_memory_in_bytes.try_into().unwrap(),
+                compressed_memory_in_bytes
+                    .try_into()
+                    .expect("compressed_memory_in_bytes should be less than 8192 PiB."),
             ),
         }
     }
 
     /// Change the amount of memory available for ingested data by `size_in_bytes`.
     pub(super) fn adjust_ingested_memory(&self, size_in_bytes: isize) {
-        // unwrap() is safe as lock() only returns an error if the mutex is poisoned.
-        *self.remaining_ingested_memory_in_bytes.lock().unwrap() += size_in_bytes;
+        *self
+            .remaining_ingested_memory_in_bytes
+            .lock()
+            .expect("Mutex should not be poisoned.") += size_in_bytes;
```

Comment:
Maybe use a variable or global constant instead of repeating the same string.

Reply:
Fixed.